go/libraries/doltcore/doltdocs/docs.go

```diff
@@ -0,0 +1,186 @@
+
+var AllValidDocDetails = &Docs{
+	{DocPk: doltdb.ReadmePk, File: ReadmeFile},
+	{DocPk: doltdb.LicensePk, File: LicenseFile},
+}
+
+func GetDocFile(filename string) string {
+	return filepath.Join(dbfactory.DoltDir, filename)
+}
+
+func LoadDocs(fs filesys.ReadWriteFS) (Docs, error) {
+	docsWithCurrentText := *AllValidDocDetails
+	for i, val := range docsWithCurrentText {
+		path := GetDocFile(val.File)
+		exists, isDir := fs.Exists(path)
+		if exists && !isDir {
+			data, err := fs.ReadFile(path)
+			if err != nil {
+				return nil, err
+			}
+			val.Text = data
+			docsWithCurrentText[i] = val
+		}
+	}
+	return docsWithCurrentText, nil
+}
+
+func (docs Docs) Save(fs filesys.ReadWriteFS) error {
+	for _, doc := range docs {
+		if !IsValidDoc(doc.DocPk) {
+			continue
+		}
+		filePath := GetDocFile(doc.File)
+		if doc.Text != nil {
+			err := fs.WriteFile(filePath, doc.Text)
+			if err != nil {
+				return err
+			}
+		} else {
+			err := DeleteDoc(fs, doc.DocPk)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func DeleteDoc(fs filesys.ReadWriteFS, docName string) error {
+	for _, doc := range *AllValidDocDetails {
+		if doc.DocPk == docName {
+			path := GetDocFile(doc.File)
+			exists, isDir := fs.Exists(path)
+			if exists && !isDir {
+				return fs.DeleteFile(path)
+			}
+		}
+	}
+	return nil
+}
+
+func IsValidDoc(docName string) bool {
+	for _, doc := range *AllValidDocDetails {
+		if doc.DocPk == docName {
+			return true
+		}
+	}
+	return false
+}
+
+func DocTblKeyFromName(fmt *types.NomsBinFormat, name string) (types.Tuple, error) {
+	return types.NewTuple(fmt, types.Uint(schema.DocNameTag), types.String(name))
+}
+
+// GetDocRow returns the associated row of a particular doc from noms.
+func GetDocRow(ctx context.Context, docTbl *doltdb.Table, sch schema.Schema, key types.Tuple) (r row.Row, ok bool, err error) {
+	rowMap, err := docTbl.GetRowData(ctx)
+	if err != nil {
+		return nil, false, err
+	}
+
+	var fields types.Value
+	fields, ok, err = rowMap.MaybeGet(ctx, key)
+	if err != nil || !ok {
+		return nil, ok, err
+	}
+
+	r, err = row.FromNoms(sch, key, fields.(types.Tuple))
+	return
+}
+
+// GetDocTextFromTbl updates the Text field of a docDetail using the provided table and schema.
+func GetDocTextFromTbl(ctx context.Context, tbl *doltdb.Table, sch *schema.Schema, doc DocDetails) (DocDetails, error) {
+	if tbl != nil && sch != nil {
+		key, err := DocTblKeyFromName(tbl.Format(), doc.DocPk)
+		if err != nil {
+			return DocDetails{}, err
+		}
+
+		docRow, ok, err := GetDocRow(ctx, tbl, *sch, key)
+		if err != nil {
+			return DocDetails{}, err
+		}
+		if ok {
+			docValue, _ := docRow.GetColVal(schema.DocTextTag)
+			doc.Text = []byte(docValue.(types.String))
+		} else {
+			doc.Text = nil
+		}
+	} else {
+		doc.Text = nil
+	}
+	return doc, nil
+}
+
+// GetDocTextFromRow updates the Text field of a docDetail using the provided row.
+func GetDocTextFromRow(r row.Row, doc *DocDetails) (DocDetails, error) {
```

Comment:
Same comment here. Don't change the value of an in param and then also return it in some cases.

Reply:
This should probably be package private.